robot/web/web.go

```diff
@@ -564,6 +566,55 @@ func (rc *RequestCounter) UnaryInterceptor(
 	return handler(ctx, req)
 }
 
+type wrappedStreamWithRC struct {
+	googlegrpc.ServerStream
+	apiMethod string
+	rc        *RequestCounter
+}
+
+func (w *wrappedStreamWithRC) RecvMsg(m any) error {
+	// Unmarshalls into m (to populate fields).
+	err := w.ServerStream.RecvMsg(m)
+
+	if err == nil && w.apiMethod != "" {
+		var key string
+		if namer, ok := m.(Namer); ok {
+			key = fmt.Sprintf("%v.%v", namer.GetName(), w.apiMethod)
+		} else {
+			key = w.apiMethod
+		}
+		if apiCounts, ok := w.rc.counts.Load(key); ok {
+			apiCounts.(*atomic.Int64).Add(1)
+		} else {
+			newCounter := new(atomic.Int64)
+			newCounter.Add(1)
+			w.rc.counts.Store(key, newCounter)
```

Comment:
this looks very similar to https://github.com/viamrobotics/rdk/pull/4941/files#diff-c53b2d6b81f57688a657b56a6d9b30f7e072d22ae5b709e39b4e05c8838bd5c4R556-R567, can we combine them into one function and call them from within unary/server interceptor?